packages/dotcom-ui-client-embed/src/client/ClientEmbedData.ts

```diff
@@ -0,0 +1,15 @@
+export default class ClientEmbedData {
+  private data
+
+  constructor(data) {
+    this.data = Object.freeze(data)
```

Comment:
Since the data in this is likely to be nested, should we consider doing a deeper freeze? may not be worth the extra code required in the bundle...